modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IORace.kt

```diff
@@ -0,0 +1,480 @@
+package arrow.fx
+
+import arrow.core.Either
+import arrow.core.Left
+import arrow.core.Right
+import arrow.core.internal.AtomicBooleanW
+import arrow.fx.extensions.io.applicative.map
+import arrow.fx.extensions.io.monad.flatMap
+import arrow.fx.internal.IOFiber
+import arrow.fx.internal.IOForkedStart
+import arrow.fx.internal.Platform
+import arrow.fx.internal.UnsafePromise
+import arrow.fx.typeclasses.Fiber
+import kotlin.coroutines.CoroutineContext
+
+interface IORace {
+
+  fun <A, B> raceN(ioA: IOOf<A>, ioB: IOOf<B>): IO<RacePair<ForIO, A, B>> =
+    IO.racePair(IODispatchers.CommonPool, ioA, ioB)
+
+  fun <A, B, C> raceN(ioA: IOOf<A>, ioB: IOOf<B>, ioC: IOOf<C>): IO<RaceTriple<ForIO, A, B, C>> =
+    IO.raceTriple(IODispatchers.CommonPool, ioA, ioB, ioC)
```

Comment:
I believe there might be a api nit here as when someone adds an extra IO to the race3 the result type will change from RaceTriple to Race4, which uses a different encoding for the fold, which is a bit annoying to be fixing, specially if you're trying out several approaches.

I suggest to forward to `racePair`/`raceTriple` if devs care about losers and just call the raceN here as well to return a Race3 (and Race2 above)